### ISSUE TITLE
0.4.9 release seems to break tokio (which depends on log)

### ISSUE THREAD

**Issue by alex** [rust-lang/log#369]
You can see an example of this here: https://github.com/alex/otp-cop/pull/374/checks?check_run_id=349720268

I didn't see a mention of breaking changes (besides MSRV) in the CHANGELOG, so I'm filing this here. Let me know if this should be filed on tokio instead though!

**Comment by mojzu** [comment]
I am receiving similar errors trying to install cargo-make.

```
   Compiling cargo-make v0.24.1
error: macro expansion ignores token `if` and any following
   --> <::log::macros::log_impl macros>:13:23
    |
13  |       let lvl = $ lvl ; if lvl <= $ crate :: STATIC_MAX_LEVEL && lvl <= $ crate
    |                         ^^
    | 
   ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/cargo-make-0.24.1/src/lib/cache.rs:106:47
    |
106 |   ...                   Err(error) => info!(
    |  _____________________________________-
107 | | ...                       "Error while writing to cache file: {:#?}, error: {:#?}",
108 | | ...                       &file_name, error
109 | | ...                   ),
    | |                       -
    | |                       |
    | |_______________________caused by the macro expansion here
    |                         help: you might be missing a semicolon here: `;`
    |
    = note: the usage of `log_impl!` is likely invalid in expression context
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

...
```

**Comment by hecrj** [comment]
It seems to break [`relevant`](https://github.com/omni-viral/relevant) and [`gfx-backend-vulkan`](https://github.com/gfx-rs/gfx/tree/master/src/backend/vulkan) too.

**Comment by pythondude325** [comment]
This also breaks [`goblin`](https://github.com/m4b/goblin)

**Comment by sfackler** [comment]
I've yanked the 0.4.9 release.

@KodrAus It looks like all of the code blocks in the log_impl macro need to be contained in a block - how did CI pass after the KV stuff landed?

**Comment by mojzu** [comment]
Install cargo-make is working again, thank you for the very quick fix!

**Comment by KodrAus** [comment]
@sfackler Hm that's a good question... I'll check this out now.

**Comment by KodrAus** [comment]
Ok, it looks like with our macro tests being brand new we didn't have coverage for the macros in expression context so CI didn't pick this up. I also totally missed this when reviewing the original implementation.

I'm putting together a PR now to fix this up.

**Comment by KodrAus** [comment]
I've just published `0.4.10` that fixes this up.
